components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/geo/service/GeoLocationProviderServiceImpl.java

```diff
@@ -157,6 +159,53 @@
         }
     }
 
+    @Override
+    public List<GeoFence> getWithinAlerts() throws GeoLocationBasedServiceException {
+
+        Registry registry = getGovernanceRegistry();
+        String registryPath = GeoServices.REGISTRY_PATH_FOR_ALERTS +
+                GeoServices.ALERT_TYPE_WITHIN + "/";
+        Resource resource;
+        try {
+            resource = registry.get(registryPath);
+        } catch (RegistryException e) {
+            log.error("Error while reading the registry path: " + registryPath);
+            return null;
```

Comment:
I think it is better to return an empty collection instead of null.

Reply:
Fixed c653051